File: steam_online_tracker.py
import requests
import time
from datetime import datetime

def get_user_info(api_key, steam_id):
    url = f"http://api.steampowered.com/ISteamUser/GetPlayerSummaries/v0002/?key={api_key}&steamids={steam_id}"
    try:
        response = requests.get(url)
        data = response.json()

        # Retrieve user information
        if 'response' in data and 'players' in data['response'] and len(data['response']['players']) > 0:
            player = data['response']['players'][0]
            return {
                'personastate': player.get('personastate'),
                'personaname': player.get('personaname'),
                'profileurl': player.get('profileurl'),
                'avatar': player.get('avatar'),
                'lastlogoff': player.get('lastlogoff'),
                'timecreated': player.get('timecreated')
            }
        else:
            print("Error retrieving user data.")
            return None
    except Exception as e:
        print(f"An error occurred: {e}")
        return None

def main():
    # Input API key and Steam ID
    api_key = input("Enter your Steam API key: ").strip()
    steam_id = input("Enter the Steam ID of the user: ").strip()
    
    online_status = None

    while True:
        user_info = get_user_info(api_key, steam_id)

        if user_info:
            current_status = user_info['personastate']
            personaname = user_info['personaname']
            profileurl = user_info['profileurl']
            avatar = user_info['avatar']
            lastlogoff = user_info['lastlogoff']
            timecreated = user_info['timecreated']

            # Convert timestamps to human-readable format
            lastlogoff_str = datetime.fromtimestamp(lastlogoff).strftime('%Y-%m-%d %H:%M:%S') if lastlogoff else 'N/A'
            timecreated_str = datetime.fromtimestamp(timecreated).strftime('%Y-%m-%d %H:%M:%S') if timecreated else 'N/A'

            # Print user information
            print(f"User: {personaname}")
            print(f"Profile URL: {profileurl}")
            print(f"Avatar URL: {avatar}")
            print(f"Last Logoff: {lastlogoff_str}")
            print(f"Account Created: {timecreated_str}")
            
            # Check if the status has changed to "online"
            if current_status == 1 and online_status != 1:
                online_status = current_status
                print(f"The user came online at {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")

        time.sleep(60)  # Check the status every 60 seconds

if __name__ == "__main__":
    main()

File: steamonlinetracker.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

class steamonlinetracker
{
    static async Task Main(string[] args)
    {
        // Input API key and Steam ID
        Console.Write("Enter your Steam API key: ");
        string apiKey = Console.ReadLine().Trim();

        Console.Write("Enter the Steam ID of the user: ");
        string steamId = Console.ReadLine().Trim();

        int? onlineStatus = null;

        while (true)
        {
            var userInfo = await GetUserInfo(apiKey, steamId);

            if (userInfo != null)
            {
                int currentStatus = userInfo.Value.personastate;
                string personaname = userInfo.Value.personaname;
                string profileurl = userInfo.Value.profileurl;
                string avatar = userInfo.Value.avatar;
                long lastlogoff = userInfo.Value.lastlogoff;
                long timecreated = userInfo.Value.timecreated;

                string lastlogoffStr = lastlogoff > 0 ? DateTimeOffset.FromUnixTimeSeconds(lastlogoff).ToString("yyyy-MM-dd HH:mm:ss") : "N/A";
                string timecreatedStr = timecreated > 0 ? DateTimeOffset.FromUnixTimeSeconds(timecreated).ToString("yyyy-MM-dd HH:mm:ss") : "N/A";

                Console.WriteLine($"User: {personaname}");
                Console.WriteLine($"Profile URL: {profileurl}");
                Console.WriteLine($"Avatar URL: {avatar}");
                Console.WriteLine($"Last Logoff: {lastlogoffStr}");
                Console.WriteLine($"Account Created: {timecreatedStr}");

                // Check if the status has changed to "online"
                if (currentStatus == 1 && onlineStatus != 1)
                {
                    onlineStatus = currentStatus;
                    Console.WriteLine($"The user came online at {DateTime.Now:yyyy-MM-dd HH:mm:ss}");
                }
            }

            await Task.Delay(60000); // Check the status every 60 seconds
        }
    }

    static async Task<(int personastate, string personaname, string profileurl, string avatar, long lastlogoff, long timecreated)?> GetUserInfo(string apiKey, string steamId)
    {
        string url = $"http://api.steampowered.com/ISteamUser/GetPlayerSummaries/v0002/?key={apiKey}&steamids={steamId}";

        try
        {
            using HttpClient client = new HttpClient();
            HttpResponseMessage response = await client.GetAsync(url);
            response.EnsureSuccessStatusCode();

            string responseBody = await response.Content.ReadAsStringAsync();
            JObject data = JObject.Parse(responseBody);

            // Retrieve user information
            var players = data["response"]?["players"];
            if (players != null && players.HasValues)
            {
                var player = players.First;

                return (
                    personastate: player.Value<int?>("personastate") ?? 0,
                    personaname: player.Value<string>("personaname"),
                    profileurl: player.Value<string>("profileurl"),
                    avatar: player.Value<string>("avatar"),
                    lastlogoff: player.Value<long?>("lastlogoff") ?? 0,
                    timecreated: player.Value<long?>("timecreated") ?? 0
                );
            }
            else
            {
                Console.WriteLine("Error retrieving user data.");
                return null;
            }
        }
        catch (Exception e)
        {
            Console.WriteLine($"An error occurred: {e.Message}");
            return null;
        }
    }
}